lib/components/Input/generic-wheel-picker/index.tsx

```diff
@@ -0,0 +1,195 @@
+
+        const handleInputClick = (e: MouseEvent<HTMLDivElement>) => {
+            onClickDropdown?.(e);
+            isOpen ? close() : open();
+        };
+
+        const handleKeyDownEvent = (e: KeyboardEvent<HTMLDivElement>) => {
+            if (e.key === "ArrowDown") {
+                open();
+            }
+            
+        };
+
+        useEffect(() => {
+            if (onValueChange) onValueChange(daataValue);
+        }, [selectedValue]);
+
+        return (
+            <>
+                <div
+                    ref={containerRef}
+                    className={clsx(
+                        "quill-generic-picker__container",
+                        `quill-generic-picker__is-open--${isOpen}`,
+                        containerClassName,
+                    )}
+                >
+                    <div ref={ref} onClick={handleInputClick} onKeyDown={handleKeyDownEvent}>
+                        <Input
+                            dropdown
+                            isDropdownOpen={isOpen}
+                            value={selectedValue}
+                            autoFocus={false}
+                            autoComplete="off"
+                            className={clsx(
+                                "quill-generic-picker__input",
+                                `quill-generic-picker__input--hasValue--${!!selectedValue}`,
+                                className,
+                            )}
+                            type="text"
+                            label={label}
+                            leftIcon={leftIcon}
+                            rightIcon={rightIcon}
+                            {...rest}
+                        />
+                    </div>
+                    <div className="quill-generic-picker__content--container">
+                        {!isMobile ? (
+                            isOpen && (
+                                <div className="quill-generic-picker__content">
+                                    <WheelPickerContainer
+                                        dataValues={daataValue}
+                                        close={close}
+                                        setSelectedValue={setSelectedValue}
+                                        setdataValues={(
+                                            index: number,
+                                            newValue: string | number,
+                                        ) => updateItemAtIndex(index, newValue)}
+                                        {...rest}
+                                    >
+                                        {children}
+                                    </WheelPickerContainer>
+                                </div>
+                            )
+                        ) : (
+                            <ActionSheet.Root
+                                isOpen={isOpen}
+                                onClose={() => close()}
```

Comment:
Will it work like this: onClose={close} ? Please, ignore if irrelevant

Reply:
yes made the change